pkg/store/prometheus.go

```diff
@@ -621,3 +649,64 @@ func (p *PrometheusStore) LabelValues(ctx context.Context, r *storepb.LabelValue
 
 	return &storepb.LabelValuesResponse{Values: m.Data}, nil
 }
+
+// seriesLabels returns the labels from Prometheus series API.
+func (p *PrometheusStore) seriesLabels(ctx context.Context, matchers []storepb.LabelMatcher, startTime, endTime int64) ([]map[string]string, error) {
+	u := *p.base
+	u.Path = path.Join(u.Path, "/api/v1/series")
+
+	q := u.Query()
+	q.Add("match[]", matchersToMetric(matchers))
+	u.RawQuery = q.Encode()
+
+	q.Add("start", string(startTime))
+	q.Add("end", string(endTime))
+
+	req, err := http.NewRequest("GET", u.String(), nil)
```

Comment:
You can use `http.NewRequestWithContext` here to shorten things :P

Reply:
Also... can we use `POST` here to avoid potential character limits with big queries?

Reply:
Oh, make sense! Thanks 😄

Reply:
I recheck the POST thing. Looks like https://github.com/prometheus/client_golang/blob/master/api/prometheus/v1/api.go also uses GET.